src/pip/_internal/utils/ui.py

```diff
@@ -407,7 +427,8 @@ def open_spinner(message):
     if sys.stdout.isatty() and logger.getEffectiveLevel() <= logging.INFO:
         spinner = InteractiveSpinner(message)
     else:
-        spinner = NonInteractiveSpinner(message)
+        # https://github.com/python/mypy/issues/1174
+        spinner = NonInteractiveSpinner(message)  # type: ignore
```

Comment:
IIUC, you can put `# type: SpinnerInterface` on the first spinner variable assignment and we'll be fine here.